distributed/stealing.py

```diff
@@ -450,7 +453,10 @@ def maybe_move_task(level, ts, sat, idl, duration, cost_multiplier):
                         thieves = _potential_thieves_for(ts, idle)
                         if not thieves:
                             continue
-                        thief = thieves[i % len(thieves)]
+                        thief = self._maybe_pick_thief(ts, thieves)
+                        if not thief:
+                            thief = thieves[i % len(thieves)]
```

Comment:
nit: might be easier to read if this was moved into `_maybe_pick_thief` (and it was no longer `maybe`).

Reply:
right, I figured this is nicer because it felt weird passing `i` to `_maybe_pick_thief`.